### PR TITLE
[WIP] feat(ui-shell): improve keyboard navigation and general a11y

### LINKED ISSUE #1497: Improve keyboard navigation and general a11y of UI shell
### Summary

Currently the UI shell is still lacking in accessible keyboard navigation patterns for some panels and menus.

### Justification

Improved usability for users

### Desired UX and success metrics

Users should be able to navigate the entire page without a mouse

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1497

This PR will introduce additional patterns for keyboard navigation and improve general a11y for UI shell

#### Changelog

**New**

- ...

**Changed**

- Prevent default Enter key behavior in header submenu

**Removed**

- ...

### CODE REVIEW

File: src/components/ui-shell/header-nav.js [hunk 1]
```diff
@@ -53,9 +63,30 @@ export default class HeaderNav extends mixin(createComponent, initComponentBySea
...
+    if (!nextHeaderSubmenuDOMNode) {
+      nextFocusTarget.focus();
+      return;
+    }
+    const startHeaderSubmenuDOMNode = start.closest('[data-header-submenu]');
+    if (!startHeaderSubmenuDOMNode) {
+      nextFocusTarget.focus();
+      return;
+    }
+    const startHeaderSubmenuClass = HeaderSubmenu.create(startHeaderSubmenuDOMNode);
```
Comment (asudoh) [inline src/components/ui-shell/header-nav.js]: I now realized that `.create()` call here make us have to manage `HeaderSubmenu` instance. We'd probably use `HeaderSubmenu.components.get()` and bail out if it returns nothing. My apologies for not noticing this earlier.

**Review by joshblack — APPROVED** [review]

**Comment by joshblack** [comment]
All good with this @emyarod?

**Comment by emyarod** [comment]
@joshblack this is a WIP branch that won't be finished until the UI shell components are merged into master. It's meant to be the PR for a11y experts to audit the UI shell work, I just opened it in advance and made some tweaks to start off

**Comment by joshblack** [comment]
cc @dakahn which keyboard navigation pattern did we end up going with? Does this PR match what we need to implement?

**Comment by dakahn** [comment]
This PR extends and elaborates on the application menu style keyboard navigation controls that are an accessibility anti-pattern for a website header nav. 🙇🏽‍♂️

I think it's totally worth revisiting this pattern conceptually especially after some user testing.

**Comment by joshblack** [comment]
@emyarod should we close this for now then as we look to switch the menu pattern back to the pattern you originally suggested? haha

**Comment by emyarod** [comment]
I think we'll need some clarification on what exactly we want to implement since we have been dealing with conflicting expectations. I don't really have a preference at this point until we have a clearly defined pattern going forward

**Comment by joshblack** [comment]
@emyarod totally fair. @dakahn do you have the strategy for the UI Shell nav documented somewhere?  I believe we can move forward with that unless anyone has an objection 👍

**Comment by dakahn** [comment]
Accessibility specification documentation for the UI-Shell is on the way for sure. Landed on a simpler interaction pattern after a fair bit of research into the latest thinking from the accessibility community. Here's a few links for reference in the mean time:

#### References
- [Don't Use Aria Menu Roles for Site Nav](http://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html)
- [Discussion on the W3C regarding expected interaction for navigation menus](https://github.com/w3c/aria-practices/issues/353)

**Comment by joshblack** [comment]
@dakahn what @emyarod is alluding to is that we brought up those same resources in the past and were told to stick with the interaction pattern that have been implemented. The concern on the implementation side is that we'll keep flip-flopping between the two sides of this argument and we want to make sure we have a good direction moving forward 👍

**Comment by dakahn** [comment]
From where I stand this is less of an argument and more of a mass misinterpretation of unclear aria authoring practices. I agree -- let's not flip flop -- let's make accessibility decisions based on the evidence of user need through iteration and testing.

Surfacing our rationale through accessibility documentation is a priority during our beta and will head off some of these controversy's and collisions.

**Comment by joshblack** [comment]
@dakahn cool, do we feel like we can proceed with this path, document it, and move forward? cc @emyarod

**Comment by emyarod** [comment]
yes that works for me

**Comment by joshblack** [comment]
Do we want to close this and refer back to this post-v10 after sprint planning this morning? cc @dakahn did you have a chance to document our UI Shell guidance somewhere?

**Pull request closed without merging**